sale_channel_hook_invoice/models/account_invoice.py

```diff
@@ -0,0 +1,28 @@
+#  Copyright (c) Akretion 2020
+#  License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+import base64
+
+from odoo import models
+
+FIELDS_SIMPLE_COPY = ["name", "amount_total_signed"]
+
+
+class AccountInvoice(models.Model):
+    _name = "account.invoice"
+    _inherit = ["account.invoice", "sale.channel.hook.mixin"]
+
+    def action_invoice_paid(self):
+        result = super().action_invoice_paid()
+        for rec in self:
+            rec.trigger_channel_hook("create_invoice")
+        return result
+
+    def get_hook_content_create_invoice(self):
```

Comment:
Only send hook for simple sale case, skip the complex case we are here for a simple implementation

sales = self.mapped("sale_line_ids.order_id")
if sales:
     content = {"sale_name": sales[0].name, "invoice": self.number}

....


else:
     return None

Reply:
done, 
+ origin determined through invoice lines->SO lines instead of "origin" field
+ presence of an origin SO is tested before triggering the hook